### ISSUE TITLE
smooth l1 loss does not match formula

### ISSUE THREAD

**Issue by tanzhenyu** [keras-team/keras-cv#920]
Here's our implementation:
https://github.com/keras-team/keras-cv/blame/master/keras_cv/losses/smooth_l1.py#L39-L42

Here's a definition used by PyTorch:
https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html

A beta denominator is missing. It still works fine if the cutoff is 1, but anything else would be different.
We should also add more unit test to it while fixing it

**Comment by bhack** [comment]
Are we talking about the same thing?

https://mohitjainweb.files.wordpress.com/2018/03/smoothl1loss.pdf

Or the self-adjusting version?
https://paperswithcode.com/method/self-adjusting-smooth-l1-loss

**Comment by tanzhenyu** [comment]
> Are we talking about the same thing?
> 
> https://mohitjainweb.files.wordpress.com/2018/03/smoothl1loss.pdf
> 
> Or the self-adjusting version? https://paperswithcode.com/method/self-adjusting-smooth-l1-loss

It seems the self adjusting one is more widely used

**Comment by bhack** [comment]
Ok as I supposed it was introduced later with Retinamask:

https://paperswithcode.com/method/retinamask

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models. Please read [the announcement](https://github.com/keras-team/keras-hub/issues/1831). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
